test/e2e/set/scyllacluster/scyllacluster_shardawareness.go

```diff
@@ -28,23 +29,29 @@ var _ = g.Describe("ScyllaCluster", func() {
 	f := framework.NewFramework("scyllacluster")
 
 	g.It("should allow to build connection pool using shard aware ports", func() {
-		g.Skip("Shardawareness doesn't work on setups NATting traffic, and our CI does it when traffic is going through ClusterIPs." +
-			"It's shall be reenabled once we switch client-node communication to PodIPs.",
-		)
-
 		const (
 			nonShardAwarePort = 9042
 			shardAwarePort    = 19042
-			nrShards          = 2
+			nrShards          = 4
 		)
 
 		ctx, cancel := context.WithTimeout(context.Background(), testTimeout)
 		defer cancel()
 
 		sc := scyllaclusterfixture.BasicScyllaCluster.ReadOrFail()
 		sc.Spec.Datacenter.Racks[0].Members = 1
+		sc.Spec.ExposeOptions = &scyllav1.ExposeOptions{
+			BroadcastOptions: &scyllav1.NodeBroadcastOptions{
+				Clients: scyllav1.BroadcastOptions{
+					Type: scyllav1.BroadcastAddressTypePodIP,
+				},
+				Nodes: scyllav1.BroadcastOptions{
+					Type: scyllav1.BroadcastAddressTypePodIP,
+				},
+			},
+		}
 
-		// Ensure 2 shards.
+		// Ensure number of shards.
 		sc.Spec.Datacenter.Racks[0].Resources.Limits[corev1.ResourceCPU] = resource.MustParse(fmt.Sprintf("%d", nrShards))
```

Comment:
Shouldn't requests also be set to the same value to have 4 cores guaranteed?

Reply:
Setting `requests` won't guarantee it either. Number of shards is controlled by the limit of the container:
https://github.com/scylladb/scylla-operator/blob/aa89840c2f7ba428d2a3e15952c6313c38302296/pkg/controller/scyllacluster/resource.go#L436-L445

https://github.com/scylladb/scylla-operator/blob/eb144ac50175c040e486f47eb9468908181ce678/pkg/sidecar/config/config.go#L237